velox/exec/TableScan.cpp

```diff
@@ -276,6 +278,7 @@ void TableScan::preload(std::shared_ptr<connector::ConnectorSplit> split) {
   // a shared_ptr to it. This is required to keep memory pools live
   // for the duration. The callback checks for task cancellation to
   // avoid needless work.
+  setInputFileName(split);
```

Comment:
Should not set file name here.  In case of preload, you can set it after the `setFromDataSource` call above